cmd/metrics-server/app/options/options.go

```diff
@@ -66,7 +65,6 @@ func (o *Options) Flags(cmd *cobra.Command) {
 
 	flags.BoolVar(&o.InsecureKubeletTLS, "kubelet-insecure-tls", o.InsecureKubeletTLS, "Do not verify CA of serving certificates presented by Kubelets.  For testing purposes only.")
 	flags.BoolVar(&o.DeprecatedCompletelyInsecureKubelet, "deprecated-kubelet-completely-insecure", o.DeprecatedCompletelyInsecureKubelet, "Do not use any encryption, authorization, or authentication when communicating with the Kubelet.")
-	flags.IntVar(&o.KubeletPort, "kubelet-port", o.KubeletPort, "The port to use to connect to Kubelets.")
```

Comment:
Not sure if we should change the default behavior. We still support `--deprecated-kubelet-completely-insecure`, I expect `node.Status.DaemonEndpoints.KubeletEndpoint.Port` to return secure port (needs verification). Meaning we will break configurations using `read-only` port. 
For now I would suggest adding additional flag like `--kubelet-use-node-status-port`

Reply:
Yes,you are right,I think adding additional flag is better. I will modify it.